### ISSUE TITLE
element does not pass a11y tests suite

### ISSUE THREAD

**Issue by tomalec** [vaadin/vaadin-date-picker#534]
### Description

I tried to test my element that uses vaadin-date-picker pass Accessibility audit via Web Components Tester, it fails due to failure for `vaadin-date-picker` itself.

![image](https://user-images.githubusercontent.com/17435/38649138-2c674fa0-3df5-11e8-894a-283c5dc70565.png)

### Expected outcome

Tests should pass

### Actual outcome

Tests fails

### Steps to reproduce
1. Put a `vaadin-date-picker` element in the test fixture
2. Set label attribute.
3. Test against WCT's `a11ySuite`

```html
<test-fixture id="a11y">
  <template>
    <vaadin-date-picker label="foo"></vaadin-date-picker>
  </template>
</test-fixture>

<script>
  a11ySuite('a11y');
</script>
```

### Browsers Affected

- [x] Chrome
- [x] Firefox
- [?] Safari
- [x] Edge
- [?] IE 11
- [?] iOS Safari
- [?] Android Chrome

**Comment by samiheikki** [comment]
It is actually vaadin-text-field element that is failing the a11y tests. Created PR: https://github.com/vaadin/vaadin-text-field/pull/252 Further discussions in there

**Comment by web-padawan** [comment]
Duplicating my comment her: https://github.com/vaadin/vaadin-text-field/pull/252#issuecomment-438964875

Actually, this check is "false positive" because of Shadow DOM unfriendliness of the `axs`.
See the lint there `axs` uses `ownerDocument.getElementById()` to check the element:

https://github.com/GoogleChrome/accessibility-developer-tools/blob/3d7c96bf34b3146f40aeb2720e0927f221ad8725/dist/js/axs_testing.js#L1456

See also GoogleChrome/accessibility-developer-tools#272

So, it needs to be tested and confirmed in the real screen readers.

**Comment by web-padawan** [comment]
Referencing the ARIA spec: https://www.w3.org/TR/wai-aria-1.1/#aria-label

> If the label text is visible on screen, authors SHOULD use aria-labelledby and SHOULD NOT use aria-label.

Confirmed by @limonte that the current approach works in NVDA, both for label and error message. Therefore, closing this issue for now as won't fix and working as intended. 

Let's reopen in case if we confirm that other screen readers which only work with `aria-label`, but not with `aria-labelledby` attribute, which is being used now.
